cpp/dolfinx/fem/utils.cpp

```diff
@@ -758,3 +760,21 @@ fem::pack_coefficients(const fem::Form& form)
   return c;
 }
 //-----------------------------------------------------------------------------
+Eigen::Array<PetscScalar, Eigen::Dynamic, 1>
+fem::pack_constants(const fem::Form& form)
+{
+  const std::vector<
+      std::pair<std::string, std::shared_ptr<const function::Constant>>>
+      constants = form.constants();
+  std::vector<PetscScalar> constant_values;
```

Comment:
Is it necessary to build a vector? Can the data go straight into the `Eigen::Array<PetscScalar, Eigen::Dynamic, 1>`?

Reply:
Not so easily, since the size of the array is not known before the loop. Std::vector here is beneficial as a resizeable storage with push_back. And the loop neither the array is will ever be big.